reinforcement_learning/include/api_status.h

```diff
@@ -0,0 +1,220 @@
+/**
+ * @brief api_status definition.  api_status is used to return error information to the caller.
+ * 
+ * @file api_status.h
+ * @author your name
+ * @date 2018-07-18
+ */
+#pragma once
+
+#include <string>
+#include <sstream>
+#include "err_constants.h"
+
+namespace reinforcement_learning {
+  /**
+   * @brief Report status of all API calls
+   */
+  class api_status {
+    public:
+      api_status();
+
+      /**
+       * @brief Get the error code 
+       * All API calls will return a status code.  If the optional api_status object is
+       * passed in, the code is set in the object also. 
+       * @return int Error code
+       */
+      int get_error_code() const;
+
+      /**
+       * @brief Get the error msg string
+       * All API calls will return a status code.  If the optional api_status object is
+       * passed in, the detailed error description is passed back using get_error_msg()
```

Comment:
it would be good to have a link to the various error codes here so a developer can quickly glance at them if he/she missed them the first time through the docs

Reply:
Done.